edx_analytics_transformers/transformers/caliper/event_transformers/video_events.py

```diff
@@ -0,0 +1,288 @@
+from edx_analytics_transformers.transformers.caliper.registry import CaliperTransformersRegistry
+
+
+EVENTS_ACTION_MAP = {
+    'load_video': 'Retrieved',
+    'play_video': 'Started',
+    'stop_video': 'Ended',
+    'complete_video': 'Ended',
+    'pause_video': 'Paused',
+    'seek_video': 'JumpedTo',
+}
+
+
+def make_video_block_id(video_id, course_id, video_block_name='video+block', block_version='block-v1'):
+    """
+    Return formatted video block id for provided video and course.
+
+    Arguments:
+        video_id        (str) : id for the video object
+        course_id       (str) : course key string
+        video_block_name(str) : video block prefix to generate video id
+        block_version   (str) : xBlock version
+
+    Returns:
+        str
+    """
+    return '{block_version}:{course_id}+type@{video_block_name}@{video_id}'.format(
+        block_version=block_version,
+        course_id=course_id,
+        video_block_name=video_block_name,
+        video_id=video_id
+    )
+
+
+def convert_seconds_to_iso(seconds):
+    """
+    Convert seconds from integer to ISO format.
+
+    Arguments:
+        seconds (int): number of seconds
+
+    Return:
+        str
+    """
+    return duration_isoformat(timedelta(
+        seconds=seconds
+    ))
+
+
+class BaseVideoTransformer(CaliperTransformer):
+    """
+    Base transformer for video interaction events.
+    """
+    type = 'MediaEvent'
+
+    def get_action(self, current_event, _):
+        """
+        Return action for the caliper event.
+
+        Arguments:
+            current_event (dict):   untransformed event
+            _             (dict):   transformed event
+
+        Returns:
+            str
+        """
+        return EVENTS_ACTION_MAP[current_event['name']]
+
+    def get_object(self, current_event, caliper_event):
+        """
+        Return object for the caliper event.
+
+        Arguments:
+            current_event (dict):   untransformed event
+            caliper_event (dict):   transformed event
+
+        Returns:
+            str
+        """
+        self.json_load_event()
+        caliper_object = caliper_event['object']
+        event = current_event['data'].copy()
+
+        course_id = current_event['context']['course_id']
+        video_id = event['id']
+
+        object_id = make_video_block_id(course_id=course_id, video_id=video_id)
+
+        caliper_object.update({
+            'id': object_id,
+            'type': 'VideoObject',
+            'duration': duration_isoformat(timedelta(
+                    seconds=event.pop('duration')
+            ))
+        })
+
+        return caliper_object
+
+    def transform(self):
+        """
+        Override the base class transform method to clean unwanted
+        fields from the transformed event.
+
+        Arguments:
+
+        Returns:
+            dict
+        """
+        transformed_event = super().transform()
+        if 'duration' in transformed_event['object']['extensions']:
```

Comment:
in that case we will have to place the code for deletion in every video event transformer and that will result in duplicated code. The idea behind putting this in base video transformer was to avoid that repetition. 
Should I remove this and add code for deleting duration in every transformer?